consensus/src/state_computer.rs

```diff
@@ -168,12 +169,18 @@ impl StateComputer for ExecutionProxy {
             }
         }
 
+        let executor = self.executor.clone();
+        let proof = finality_proof.clone();
         monitor!(
             "commit_block",
-            self.executor
-                .commit_blocks(block_ids, finality_proof.clone())
-                .expect("Failed to commit blocks")
-        );
+            tokio::task::spawn_blocking(move || {
+                executor
+                    .commit_blocks(block_ids, proof)
+                    .expect("Failed to commit blocks");
+            })
+            .await
```

Comment:
If you are awaiting here (i.e. blocking), what is the point of spawning to another thread here?

Reply:
the idea is to yield the cooperative  thread to other async tasks to use and spawn to dedicated blocking threads.

imagine if we only have 1 worker thread, it'd block all consensus tasks until it finished before but now it'll yield the thread back

Reply:
awaiting is not exactly the same as blocking sync function, it's more cooperative

Reply:
Ah, it's cooperative to other work,  not to  this task. 


How many distinct tasks do we have? 
Round manager,  buffer manager,  execution?

Reply:
for consensus I think we have 8 tasks. network, epoch manager, round manager, buffer manager, execution, signing, commit, quorum store (no-op today).

Reply:
it actually seems improving performance on forge, what I see on other pr is all up : 5591 TPS, 3055 ms latency, 5050 ms p99 latency,no expired txns

Reply:
Seems like the latest run did `5847 TPS` but the one previous to that did ` 6430 TPS`. Did anything change between these two runs?

Reply:
I only rebased on main